### ISSUE TITLE
Incorrect violation for `jinja: obj must be a list of dicts or a nested dict` in loop with filters

### ISSUE THREAD

**Issue by nre-ableton** [ansible/ansible-lint#2464]
##### Summary

In ansible-lint 6.6.0, the `jinja[invalid]` violation is incorrectly raised for Jinja2 loop expressions with subfilters.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible [core 2.13.4]
  ...snip...
  python version = 3.10.6 (main, Sep 14 2022, 13:37:48) [GCC 11.2.0]
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version

ansible-lint 6.6.0 using ansible 2.13.4
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Example tasks to reproduce the issue:

```yaml
- name: Query GitHub for release artifacts
  ansible.builtin.uri:
    url: "{{ release_url }}/{{ item.item.release_tag }}"
    headers:
      Accept: "application/json"
      Authorization: "token {{ release_download_credential }}"
    return_content: true
  register: github_release_query
  # Note: this loop is fine, and ansible-lint doesn't mind it
  loop: "{{ artifact_versions_dict | flatten(levels=1) }}"

# Using subelements flattens out the nested artifact list. This allows us to iterate
# through a single loop and only download when the filename of the artifact matches the
# requested filename.
- name: Download artifacts
  ansible.builtin.get_url:
    url: "{{ item.1.url }}"
    checksum: "{{ item.0.item.item.package_checksum }}"
    dest: "{{ artifact_dest }}/{{ item.1.name }}"
    headers:
      Accept: "application/octet-stream"
      Authorization: "token {{ release_download_credential }}"
    mode: "0644"
  # This line causes ansible-lint 6.6.0 to complain with:
  # jinja: obj must be a list of dicts or a nested dict (jinja[invalid])
  loop: "{{ github_release_query.results | subelements('json.assets', {'skip_missing': True}) }}"
```

**Comment by nre-ableton** [comment]
@ssbarnea thanks for the quick fix!
